sources/packages/backend/apps/queue-consumers/src/processors/schedulers/ier-Integrations/ier-integration.scheduler.ts

```diff
@@ -0,0 +1,75 @@
+import { InjectQueue, OnQueueActive, Process, Processor } from "@nestjs/bull";
+import { IER12FileService } from "@sims/integrations/institution-integration/ier-integration";
+import {
+  IER_SCHEDULER_JOB_ID,
+  QUEUE_RETRY_DEFAULT_CONFIG,
+} from "@sims/services/constants";
+import { QueueNames } from "@sims/services/queue";
+import { PST_TIMEZONE } from "@sims/utilities";
+import { ConfigService } from "@sims/utilities/config";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import Bull, { CronRepeatOptions, Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import {
+  GeneratedDateQueueInDTO,
+  IER12ResultQueueOutDTO,
+} from "./models/ier.model";
+
+@Processor(QueueNames.IERIntegration)
+export class IERIntegrationScheduler extends BaseScheduler<GeneratedDateQueueInDTO> {
+  cronOptions: Bull.JobOptions = undefined;
+
+  constructor(
+    @InjectQueue(QueueNames.IERIntegration)
+    readonly schedulerQueue: Queue<GeneratedDateQueueInDTO>,
+    private readonly ierRequest: IER12FileService,
+    config: ConfigService,
+  ) {
+    super();
+    // cron options.
+    this.cronOptions = {
+      ...QUEUE_RETRY_DEFAULT_CONFIG,
+      jobId: IER_SCHEDULER_JOB_ID,
+      repeat: {
+        cron: config.queueSchedulerCrons.ierCron,
+        tz: PST_TIMEZONE,
+      } as CronRepeatOptions,
+    };
+  }
+
+  /**
+   * Add scheduler to the queue.
+   */
+  async initializeScheduler(): Promise<void> {
+    await this.schedulerQueue.add(undefined, this.cronOptions);
+  }
+
+  /**
+   * Identifies all the applications which are in assessment
+   * for a particular institution and generate the request file.
+   * @params job has generatedDate Date in which the assessment for
+   * particular institution is generated.
+   * @returns Processing result log.
+   */
+  @Process()
+  async processIER12File(
+    job: Job<GeneratedDateQueueInDTO | undefined>,
+  ): Promise<IER12ResultQueueOutDTO[]> {
+    this.logger.log("Executing IER 12 file generation ...");
+    const uploadResult = await this.ierRequest.processIER12File(
+      job.data.generatedDate,
+    );
+    this.logger.log("IER 12 file generation completed.");
+    return uploadResult;
+  }
+
+  @OnQueueActive()
```

Comment:
This event is basically to inform other process which needs to know if the job is started. 
for e.g if we want to send an email to operations team whenever this job starts, we use these events. 
If it is just logging, then line 58 is as good as line 68.

Reply:
mm, I don't think it only informs another process. This is a local event handler, which is invoked whenever the local event is emitted. I don't think it wrong to add logs and these are logs are job info. 
I believe to trigger another process, I think we need to use global event listers.
Anyways, I don't know if it's necessary to add job logs here, I added them as I was exploring this area, and I think I found the logging in the nest doc too. 
If the devs think that it's not required/useless here, I can remove it. thoughts?
https://docs.nestjs.com/techniques/queues#event-listeners

Reply:
Yes I know that It is a local event handler. My point it what is the significance using this event to log alone? when listening to any event, IMO there must be a valid justification.

Reply:
as I mentioned above, I added them as I was exploring this area.  moved the job to the process